### ISSUE TITLE
Error adding node with `eks` flavor

### ISSUE THREAD

**Issue by richardcase** [kubernetes-sigs/cluster-api-provider-aws#1954]
/kind bug

**What steps did you take and what happened:**
Created a kind cluster, did `tilt up` with a tilt config that enabled the aws and eks-bootstratp providers. Then used the [cluster-template-eks.yaml](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/templates/cluster-template-eks.yaml) to create a workload cluster.

The EKS cluster is created and other AWS resources like the EC2 instance for the worker node. The `AWSMachine` is in a `running` state. However, there is an error reported reported by the capi controller:

```bash
[manager] E0918 13:20:09.701862       8 machine_controller.go:267] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="cannot assign NodeRef to Machine \"capi-managed-test-md-0-9f94564f4-64bv9\" in namespace \"default\", no matching Node: requeue in 20s" "cluster"="capi-managed-test" "machine"="capi-managed-test-md-0-9f94564f4-64bv9" "namespace"="default" 
```

And then if you use the `user` kubeconfig to query the EKS cluster there is no node `kubectl --kubeconfig=./managed-test.kubeconfig get nodes`

**What did you expect to happen:**
I expect there to be no error and the node to be listed when querying.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by richardcase** [comment]
Initial issue is that the `nodes.cluster-api-provider-aws.sigs.k8s.io` policy needs ` eks:DescribeCluster` permission to the cluster.

**Comment by richardcase** [comment]
I have this working manually and will add code to fix

**Comment by richardcase** [comment]
As well as the IAM issue this is caused by the `aws-auth` configmap not existing and therefore the kubelet isn't authorized to join. It needs to exist with a mapping for each node. For example:
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: aws-auth
  namespace: kube-system
data:
  mapRoles: |
    - rolearn: arn:aws:iam::12345678:role/nodes.cluster-api-provider-aws.sigs.k8s.io
      username: system:node:ip-10-0-115-239.eu-west-2.compute.internal
      groups:
        - system:bootstrappers
        - system:nodes
```
This part will be fixed by doing  #1850

**Comment by richardcase** [comment]
The fix for this will pave the way to help implement #1850...which will be done as a separate PR after this has merged.
